File: JAVA/C++/Basic_Prac/countDigits.cpp
#include <iostream>
using namespace std;
int main()
{
    int num, count = 0;
    cout << "Enter no: " << endl;
    cin >> num;
    if (num == 0)
    {
        count = 1;
    }
    else
    {
        while (num > 0)
        {
            count++;
            num = num / 10;
        }
    }
    cout << "end " << count;
    return 0;
}

File: JAVA/JAVA/Practice/countDigits.java
import java.util.Scanner;

public class countDigits {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter a number:");
        int n = sc.nextInt();
        int num = n;

        System.out.println("Digits from last to first:");
        while(num!=0) {
            int lastDigit = num % 10;
            num = num / 10; // reduces number
            System.out.println(lastDigit); // prints one digit
        }

        sc.close();
    }
}
